packages/lesswrong/components/common/ForumDropdownMultiselect.tsx

```diff
@@ -0,0 +1,167 @@
+    width: 19,
+    height: 19,
+  },
+  menu: {
+    marginTop: 28,
+    ...(isEAForum && {
+      "& a:hover": {
+        opacity: "inherit",
+      },
+    }),
+    '& .MuiList-padding': {
+      padding: '4px 0px',
+    }
+  },
+  menuItem: {
+    ...(isEAForum && {
+      color: theme.palette.grey[1000],
+      borderRadius: theme.borderRadius.small,
+      padding: "6px 8px",
+      margin: "0px 4px",
+      fontSize: "14px",
+      lineHeight: "14px",
+      "&:hover": {
+        backgroundColor: theme.palette.grey[250],
+        color: theme.palette.grey[1000],
+      },
+    }),
+    "&:focus": {
+      outline: "none",
+    },
+  },
+});
+
+const ForumDropdownMultiselect = ({values, options, queryParam, onSelect, eventProps, classes, className}:{
+  values: string[],
+  options: Record<string, SettingsOption>,
+  queryParam?: string,
+  onSelect?: (value: string) => void,
+  eventProps?: Record<string, string>
+  classes: ClassesType,
+  className?: string,
+}) => {
+  const [anchorEl, setAnchorEl] = useState<HTMLElement | null>(null)
+  const { captureEvent } = useTracking()
+  const label = values.reduce((prev, next) => {
+    const nextLabel = options[next].shortLabel || options[next].label
+    if (!prev) return nextLabel
+    return `${prev}, ${nextLabel}`
+  }, '')
+  const { MenuItem, ForumIcon } = Components;
+
+  const dropdownIcon = isEAForum ? <ForumIcon icon="ThickChevronDown" className={classes.dropdownIcon} /> : <ArrowDropDownIcon className={classes.dropdownIcon}/>
+  return (
+    <div className={classNames(classes.root, className)}>
+      <Button
+        variant="contained"
+        onClick={(e) => {
+          captureEvent("ForumDropdownMultiselectOpened", {value: values, ...eventProps})
+          setAnchorEl(e.currentTarget)
+        }}
+        className={classNames(classes.button, { [classes.openButton]: Boolean(anchorEl) })}
+      >
+        {label} {dropdownIcon}
+      </Button>
+      <Menu open={Boolean(anchorEl)} anchorEl={anchorEl} onClose={() => setAnchorEl(null)} className={classes.menu}>
+        {Object.keys(options).map((option) => {
+          const menuItem = <MenuItem
+            key={option}
+            value={option}
+            onClick={() => {
+              captureEvent("ForumDropdownMultiselectItemSelected", {oldValue: values, newValue: option, ...eventProps})
+              setAnchorEl(null);
+              onSelect?.(option);
+            }}
+            className={classes.menuItem}
+          >
+            {options[option].label}
+            {values.includes(option) && isEAForum && (
+              <>
+                <div className={classes.padding}></div>
+                <ForumIcon icon="Check" className={classes.selectedIcon} />
+              </>
+            )}
+          </MenuItem>
+          
+          if (queryParam) {
+            return <QueryLink key={option} query={{ [queryParam]: option }} merge>
+              {menuItem}
+            </QueryLink>
+          }
```

Comment:
Here I pulled out the `QueryLink` wrapper, since I don't want it for my other PR